lock-api/src/test/java/com/palantir/lock/client/IdentifiedLockRequestTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+public class IdentifiedLockRequestTest {
+    private static final String SERIALIZED_LOCK_REQUEST = "{"
+            + "\"requestId\":\"885afd9c-de62-44ff-a517-5db14b71bfaa\","
+            + "\"lockDescriptors\":[{\"bytes\":\"Zm9v\"}],"
+            + "\"acquireTimeoutMs\":123,"
+            + "\"clientDescription\":\"Thread: main\"}";
+
+    private static final ObjectMapper mapper = new ObjectMapper()
+            .registerModule(new Jdk8Module())
+            .registerModule(new GuavaModule());
+
+    @Test
+    public void serializationDeserialization() throws Exception {
+        IdentifiedLockRequest request = mapper.readValue(SERIALIZED_LOCK_REQUEST, IdentifiedLockRequest.class);
+        assertThat(mapper.writeValueAsString(request)).isEqualTo(SERIALIZED_LOCK_REQUEST);
+    }
```

Comment:
I realise the implementation works, but I'd prefer we test this a different way, because comparing string equality imposes an ordering on the fields, which I don't think is guaranteed by an ObjectMapper without specific annotations (e.g. http://fasterxml.github.io/jackson-annotations/javadoc/2.3.0/com/fasterxml/jackson/annotation/JsonPropertyOrder.html)

Reply:
why do we care about serialising the old thing? we're not doing that anymore right? Isn't it sufficient to just check that the `SERIALIZED_LOCK_REQUEST` *deserialises* to something that we can understand and expect? Unless I'm mistaken

Reply:
problem is you need to upgrade atlas versions on both the client and the timelock server, which is not possible. So we need to make sure we can serialize and deserialize from old to new version (old atlas client, new timelock server), and vice versa (old timelock server, new atlas client)